src/forms/labels/field-set-legend.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import { convertNameToLabel } from '../helpers'
+
+/**
+ *
+ * A legend representing a caption for the content of its parent field set element
+ *
+ * This component must be used as a direct child and the only legend of the <fieldset> element that groups related controls
+ * 
+ *
+ * The text of the legend is set using the following rules:
+ * - If the `label` prop is set to `false`, the legend is hidden visually
+ * - Else If the `label` prop is set to a string, the label will display that text
+ * - Otherwise, the label will be set using the `name` prop.
+ *
+ * 
+ * @name FieldSetLegend
+ * @type Function
+ * @param {String} name - The name of the associated group
+ * @param {String} [hint] - A usage hint for the associated input
+ * @param {String|Boolean} [label] - Custom text for the legend
+ * @param {Boolean} [required] - A boolean value to indicate whether the field is required
+ * @param {String} [requiredIndicator] - Custom character to denote a field is required (optional, default `''`)
```

Comment:
You don't need to specify `optional, default` in the description. `optional` is communicated already using the brackets (i.e., `[]`) and you can set the default by using `=''`.

Here's an example: [input](https://github.com/LaunchPadLab/lp-components/blob/052921a7cb7642867ce76ef4ea12a5f2a62b9144/src/forms/inputs/range-input.js#L25) and  [output](https://github.com/LaunchPadLab/lp-components/blob/main/docs.md#rangeinput)